frontend/public/components/utils/storage-class-dropdown.tsx

```diff
@@ -0,0 +1,206 @@
+      if (!state.defaultClass) {
+        // Add No Storage Class option if there is not a default storage class
+        unorderedItems[''] = { kindLabel: '', name: noStorageClass };
+      }
+
+      if (!this.props.loaded || !selectedKey ) {
+        state.title = <span className="text-muted">{nextProps.placeholder}</span>;
+      }
+
+      if (selectedKey) {
+        const item = unorderedItems[selectedKey];
+        if (item) {
+          state.title = this.getTitle(item.kindLabel, item.name);
+        }
+      }
+
+      if (!selectedKey && state.defaultClass) {
+        const item = unorderedItems[state.defaultClass];
+        if (item) {
+          state.title = this.getTitle(item.kindLabel, item.name);
+        }
+      }
+
+      Object.keys(unorderedItems).sort().forEach(function(key) {
+        state.items[key] = unorderedItems[key];
+      });
+      this.setState(state);
+    }
+
+    componentDidUpdate() {
+      const {defaultClass, selectedKey } = this.state;
+      if (!selectedKey && defaultClass) {
+        this.onChange(defaultClass);
+      }
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+      return !_.isEqual(this.state, nextState);
+    }
+
+    autocompleteFilter = (text, item) => fuzzy(text, item.props.name);
+
+    getTitle = (kindLabel, name) => {
+      return kindLabel ? <ResourceName kind="StorageClass" name={name} /> : <span >{name}</span>;
+    }
+
+    onChange = key => {
+      const { name, kindLabel, accessMode } = _.get(this.state, ['items', key], {});
+      this.setState({
+        selectedKey: key,
+        title: this.getTitle(kindLabel, name),
+      });
+      this.props.onChange(key, accessMode);
+    };
+
+    render() {
+      const { placeholder, id, loaded } = this.props;
+      const items = {};
+      _.each(
+        this.state.items,
+        (props, key) =>
+          (items[key] = key ? <StorageClassDropdownEntry {...props} /> : <StorageClassDropdownNoStorageClassOption {...props} />)
+      );
+
+      const { selectedKey, defaultClass } = this.state;
+
+      const Component = <Dropdown
+        autocompleteFilter={this.autocompleteFilter}
+        autocompletePlaceholder={placeholder}
+        items={items}
+        selectedKey={selectedKey}
+        title={this.state.title}
+        onChange={this.onChange}
+        id={id}
+        menuClassName="dropdown-menu--text-wrap"
+      />;
+      // Only show the dropdown if 'no storage class' is not the only option which depends on defaultClass
+      const itemsAvailableToShow = (_.size(items)>1 && !defaultClass) || (_.size(items)>0 && defaultClass);
+      const helpMes = defaultClass ? `Default storage class ${defaultClass} will be used if no other storage class selected.` : 'Optional storage class for the new claim.';
+      return <React.Fragment>
+        {loaded && itemsAvailableToShow &&
+          <div>
+            <label className="control-label" htmlFor="storageclass-dropdown">
```

Comment:
Should this use `id`?

```suggestion
            <label className="control-label" htmlFor={id}>
```